[Profile] Replace `SIGTERM` with `SIGQUIT`

giordano opened the pull request:
The `SIGQUIT` signal is handled specially in CI and automatically uploads failure information artifacts, which may be helpful to debug failures.

giordano committed:
[Profile] Replace `SIGTERM` with `SIGQUIT`

The `SIGQUIT` signal is handled specially in CI and automatically uploads
failure information artifacts, which may be helpful to debug failures.

vtjnash commented on stdlib/Profile/test/runtests.jl:
```diff
@@ -172,7 +172,7 @@ let cmd = Base.julia_cmd()
         println("KILLING debuginfo registration test BY PROFILE TEST WATCHDOG\n")
-        kill(p, Base.SIGTERM)
+        kill(p, Base.SIGQUIT)
         sleep(10)
```
```suggestion
        sleep(30)
        kill(p, Base.SIGQUIT)
        sleep(30)
```

vtjnash commented on stdlib/Profile/test/runtests.jl:
```diff
@@ -202,7 +202,7 @@ if Sys.isbsd() || Sys.islinux()
                 println("KILLING siginfo/sigusr1 test BY PROFILE TEST WATCHDOG\n")
-                kill(p, Base.SIGTERM)
+                kill(p, Base.SIGQUIT)
                 sleep(10)
```
```suggestion
                sleep(30)
                kill(p, Base.SIGQUIT)
                sleep(30)
```

vtjnash reviewed (approved):
LGTM. Even better though: let's do a double-kill, as the first one tries to print failure information (which might hang) but also first uninstalls the SIGQUIT handler, so the second one will likely run with SIG_DFL and end the process in the kernel (if the first one got stuck before doing exactly that itself)

giordano committed:
Apply suggestions from code review

Do a double-kill, as the first one tries to print failure information (which might hang) but also first uninstalls the SIGQUIT handler, so the second one will likely run with SIG_DFL and end the process in the kernel (if the first one got stuck before doing exactly that itself)

Co-authored-by: Jameson Nash <vtjnash@gmail.com>

giordano commented:
@vtjnash somewhat good news: the profile tests failed on the merge commit of this PR: https://buildkite.com/julialang/julia-master/builds/39449#019180ea-6aeb-4981-beb7-0bf4f50c0722/848-1633. Where do we find the artifacts?

vtjnash commented:
No core files appear to have been detected after the run exited
